server/src/main/java/io/unitycatalog/server/service/TableService.java

```diff
@@ -2,55 +2,168 @@
+
+  @SneakyThrows
+  public TableService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createTable(CreateTable createTable) {
+  // The Databricks API does not have a create table endpoint, so defining sensible rules here.
+  @AuthorizeExpression("""
+      #authorize(#principal, #metastore, OWNER) ||
+      #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA)
+      """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse createTable(
+      @AuthorizeKeys({
+            @AuthorizeKey(value = SCHEMA, key = "schema_name"),
+            @AuthorizeKey(value = CATALOG, key = "catalog_name")
+          })
+          CreateTable createTable) {
     assert createTable != null;
-    TableInfo createTableResponse = TABLE_REPOSITORY.createTable(createTable);
-    return HttpResponse.ofJson(createTableResponse);
+    TableInfo tableInfo = TABLE_REPOSITORY.createTable(createTable);
+    initializeAuthorizations(tableInfo);
+    return HttpResponse.ofJson(tableInfo);
   }
 
   @Get("/{full_name}")
-  public HttpResponse getTable(@Param("full_name") String fullName) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse getTable(@Param("full_name") @AuthorizeKey(TABLE) String fullName) {
     assert fullName != null;
     TableInfo tableInfo = TABLE_REPOSITORY.getTable(fullName);
     return HttpResponse.ofJson(tableInfo);
   }
 
   @Get("")
+  @AuthorizeExpression("#defer")
   public HttpResponse listTables(
       @Param("catalog_name") String catalogName,
       @Param("schema_name") String schemaName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken,
       @Param("omit_properties") Optional<Boolean> omitProperties,
       @Param("omit_columns") Optional<Boolean> omitColumns) {
-    return HttpResponse.ofJson(
-        TABLE_REPOSITORY.listTables(
+
+    ListTablesResponse listTablesResponse = TABLE_REPOSITORY.listTables(
             catalogName,
             schemaName,
             maxResults,
             pageToken,
             omitProperties.orElse(false),
-            omitColumns.orElse(false)));
+            omitColumns.orElse(false));
+
+    filterTables("""
+            #authorize(#principal, #metastore, OWNER) ||
+            (#authorizeAll(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG)) ||
+            (#authorize(#principal, #schema, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
```

Comment:
Similar rules as getTable to list tables

- catalog owner
- schema owner with USE_CATALOG
- principal with USE_CATALOG, USE_SCHEMA, and any privilege on the table